lib/SILOptimizer/Transforms/PruneVTables.cpp

```diff
@@ -27,10 +27,16 @@ using namespace swift;
 
 namespace {
 class PruneVTables : public SILModuleTransform {
-  void runOnVTable(SILModule *M,
-                   SILVTable *vtable) {
+  void runOnVTable(SILModule *M, SILVTable *vtable) {
     LLVM_DEBUG(llvm::dbgs() << "PruneVTables inspecting table:\n";
                vtable->print(llvm::dbgs()));
+    if (!M->isWholeModule() &&
+        vtable->getClass()->getEffectiveAccess() >= AccessLevel::FilePrivate) {
```

Comment:
Shouldn't it be `>` not `>=` then?

Reply:
Oops, I thought Id fixed that.  #33808

Reply:
I think this change is unnecessarily pessimistic. The check for `calleesAreStaticallyKnowable ` [below](https://github.com/apple/swift/pull/33770/files#diff-fa3907ca639ee730d10b9c0a357f10d2L76) should already prevent us from pruning nonfinal internal vtable entries. We should still be able to prune entries for final methods as long as the final method is not itself an override, because it should never need vtable dispatch to begin with.

Reply:
It looks like if a method is originally final we already don't give it a vtable entry at all. So, we should be able to remove the `if (!methodDecl->isFinal())` precondition below and get the same effect. I think that's better than reimplementing `calleesAreStaticallyKnowable` here.